packages/core/echo/echo/src/internal/types/typename.ts

```diff
@@ -0,0 +1,104 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import type * as Schema from 'effect/Schema';
+
+import { invariant } from '@dxos/invariant';
+import { DXN } from '@dxos/keys';
+
+import { getSchemaTypename } from '../annotations';
+
+import { type AnyProperties } from './base';
+
+/**
+ * Property name for typename when object is serialized to JSON.
+ */
+export const ATTR_TYPE = '@type';
+
+/**
+ * DXN to the object type.
+ */
+export const TypeId = Symbol.for('@dxos/echo/Type');
```

Comment:
I think it was better when the symbols were together with the object model definition in the `model.ts`